apps/dashboard/app/(app)/ratelimits/[namespaceId]/logs/filters.schema.ts

```diff
@@ -134,11 +32,50 @@ export const filterFieldConfig: FilterFieldConfigs = {
     type: "string",
     operators: ["is"],
     validValues: ["blocked", "passed"],
-    getColorClass: (value) => {
-      if (value === "blocked") {
-        return "bg-warning-9";
-      }
-      return "bg-success-9";
-    },
+    getColorClass: (value) => (value === "blocked" ? "bg-warning-9" : "bg-success-9"),
   } as const,
 };
+
+// Schemas
+export const ratelimitFilterOperatorEnum = z.enum(["is", "contains"]);
+export const ratelimitFilterFieldEnum = z.enum([
+  "startTime",
+  "endTime",
+  "since",
+  "identifiers",
+  "requestIds",
+  "status",
+]);
+export const filterOutputSchema = createFilterOutputSchema(
+  ratelimitFilterFieldEnum,
+  ratelimitFilterOperatorEnum,
+  ratelimitFilterFieldConfig,
+);
+
+// Types
+export type RatelimitFilterOperator = z.infer<typeof ratelimitFilterOperatorEnum>;
+export type RatelimitFilterField = z.infer<typeof ratelimitFilterFieldEnum>;
+
+export type FilterFieldConfigs = {
+  startTime: NumberConfig<RatelimitFilterOperator>;
+  endTime: NumberConfig<RatelimitFilterOperator>;
+  since: StringConfig<RatelimitFilterOperator>;
+  identifiers: StringConfig<RatelimitFilterOperator>;
+  requestIds: StringConfig<RatelimitFilterOperator>;
+  status: StringConfig<RatelimitFilterOperator>;
+};
+
+export type RatelimitFilterUrlValue = Pick<
+  FilterValue<RatelimitFilterField, RatelimitFilterOperator>,
+  "value" | "operator"
+>;
+export type RatelimitFilterValue = FilterValue<RatelimitFilterField, RatelimitFilterOperator>;
+
+export type RatelimitQuerySearchParams = {
+  startTime?: number | null;
```

Comment:
do we really need undefined and null?
I have no idea, just curious

Reply:
This is needed for nuqs's default value. It's weird sometimes its complaining for null sometimes for undefined, so I just dumped both them there 😄